tg_react/api/accounts/views.py

```diff
@@ -0,0 +1,155 @@
+
+from django.utils.translation import ugettext as _
+
+from settings.base import SITE_URL
+
+
+class UnsafeSessionAuthentication(SessionAuthentication):
+    def enforce_csrf(self, request):
+        pass
+
+
+class UserDetails(generics.RetrieveAPIView):
+    serializer_class = UserDetailsSerializer
+    authentication_classes = (SessionAuthentication, )
+    permission_classes = (IsAuthenticated, )
+
+    def get_object(self):
+        return self.request.user
+
+
+class AuthenticationView(APIView):
+    class UnsafeSessionAuthentication(SessionAuthentication):
+        def enforce_csrf(self, request):
+            pass
+
+    throttle_classes = ()
+    permission_classes = (AllowAny,)
+    authentication_classes = (UnsafeSessionAuthentication, )
+    serializer_class = AuthenticationSerializer
+
+    def post(self, request):
+        serializer = self.serializer_class(data=request.DATA)
+        if serializer.is_valid():
+            from django.contrib.auth import login
+            login(request, serializer.user)
+
+            return Response({'success': True})
+
+        return Response({'errors': serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class LogoutView(APIView):
+    throttle_classes = ()
+    permission_classes = (IsAuthenticated, )
+    authentication_classes = (SessionAuthentication, )
+
+    def post(self, request):
+        from django.contrib.auth import logout
+        logout(request)
+
+        return Response({'success': True})
+
+
+class SignUpView(APIView):
+    serializer_class = SSerializer
+    permission_classes = (AllowAny, )
+    authentication_classes = (UnsafeSessionAuthentication, )
+
+    def post(self, request):
+        serializer = self.serializer_class(data=request.DATA, context={'request': request})
+        if serializer.is_valid():
+            data = serializer.validated_data.copy()
+            password = data.pop('password', None)
+
+            user = get_user_model()(**data)
+            user.set_password(password)
+            user.save()
+
+            from django.contrib.auth import login, authenticate
+            user = authenticate(email=data['email'], password=password)
+            login(request, user)
+
+            return Response({'success': True})
+
+        return Response({'errors': serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ForgotPassword(APIView):
+    """
+    Initiate a password restore procedure.
+    """
+
+    # Example request data
+    # {
+    #   "email" : "chuvak2353@gmail.com"
+    # }
+
+    serializer_class = ForgotPasswordSerializer
+    authentication_classes = (UnsafeSessionAuthentication, )
+    permission_classes = (AllowAny, )
+
+    def send_email_notification(self, user, uid_and_token_b64):
+        # define email notification logic
+        from django.core.mail import EmailMultiAlternatives
+        # make confirm reset url
+        confirm_reset_url = '%s/reset_password/%s' % (SITE_URL, uid_and_token_b64)
```

Comment:
This pattern should come from settings, so we can add it to webpack constants. Then we can access it in the frontend and make the route based on the value